user/tlu/module/src/TluRawEvent2StdEventConverter.cc

```diff
@@ -29,41 +30,126 @@ bool TluRawEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::Standa
     d2->SetTag(TLU+stm+"_TRG", std::to_string(d1->GetTriggerN()));
   }
 
-  auto triggersFired = std::stoi(d1->GetTag("TRIGGER" , "0"), nullptr, 2); // interpret as binary
-  auto finets0 = std::stoi(d1->GetTag("FINE_TS0", "0"));
-  auto finets1 = std::stoi(d1->GetTag("FINE_TS1", "0"));
-  auto finets2 = std::stoi(d1->GetTag("FINE_TS2", "0"));
-  auto finets3 = std::stoi(d1->GetTag("FINE_TS3", "0"));
-  auto finets4 = std::stoi(d1->GetTag("FINE_TS4", "0"));
-  auto finets5 = std::stoi(d1->GetTag("FINE_TS5", "0"));
+  uint8_t triggersFired;
+  uint32_t finets0;
+  uint32_t finets1;
+  uint32_t finets2;
+  uint32_t finets3;
+  uint32_t finets4;
+  uint32_t finets5;
 
-  // Set times for StdEvent in picoseconds (timestamps provided in nanoseconds):
-  // Note: __builtin_popcount counts the "ones" in a binary, see here:
-  // https://www.geeksforgeeks.org/count-set-bits-in-an-integer/
-  // I.e. add up all timestamps for which the triggersFired bit is 1 and divide by number of active triggers:
-  // Factor 25./32. -> fine timestamp in 781ps binning
-  auto finets = ((finets0 * ((triggersFired >> 0) & 0x1))
-               + (finets1 * ((triggersFired >> 1) & 0x1))
-               + (finets2 * ((triggersFired >> 2) & 0x1))
-               + (finets3 * ((triggersFired >> 3) & 0x1))
-               + (finets4 * ((triggersFired >> 4) & 0x1))
-               + (finets5 * ((triggersFired >> 5) & 0x1)))
-               / __builtin_popcount(triggersFired & 0x3F); // count "ones" in binary
-
-  auto ts = (d1->GetTimestampBegin() * 1000) & 0xFFFFFFFFFFFFFF00 + static_cast<uint64_t>(25. / 32. * finets);
+  // Should I read these in at m_first_time and store in static member variable or read in each time?
+  double tof_scint0 = conf->Get("tof_scint0", 0.); // in nanoseconds
+  double tof_scint1 = conf->Get("tof_scint1", 0.); // in nanoseconds
+  double tof_scint2 = conf->Get("tof_scint2", 0.); // in nanoseconds
+  double tof_scint3 = conf->Get("tof_scint3", 0.); // in nanoseconds
+  double tof_scint4 = conf->Get("tof_scint4", 0.); // in nanoseconds
+  double tof_scint5 = conf->Get("tof_scint5", 0.); // in nanoseconds
```

Comment:
input parameters in ns (without units) or in 780ps bins: up for discussion, I'd still opt for ns

Reply:
Changed to `delay_scint<i>` in 781.25ps bins now, README updated accordingly.

Also tested 3 cases, see screenshot below:
1) all delays zero
2) delays = geometric time-of-flight
3) delays = measured offsets in fineTS0 vs. fineTS1 histograms

As expected, 3) gives the best results.
The only problem left is: I extracted the delay values from the histograms fineTS0 vs. fineTS1 etc. in corryvreckan, see [here](https://gitlab.cern.ch/jekroege/corryvreckan/-/blob/lab_analysis/src/modules/EventLoaderEUDAQ2/EventLoaderEUDAQ2.cpp#L99).
in the `EventLoaderEUDAQ2`. 
Since it's not generic at all, I'm lacking an idea how to make this available in the master. Maybe a new module...?

![image](https://user-images.githubusercontent.com/47758962/93211496-9ca26c00-f761-11ea-9de5-21dddeb301bb.png)